packages/components/src/popover/index.js

```diff
@@ -33,226 +33,194 @@ const FocusManaged = withConstrainedTabbing( withFocusReturn( ( { children } ) =
-		clearTimeout( this.focusTimeout );
-		this.toggleAutoRefresh( false );
-	}
-
-	toggleAutoRefresh( isActive ) {
-		window.cancelAnimationFrame( this.rafHandle );
-
-		// Refresh the popover every time the window is resized or scrolled
-		const handler = isActive ? 'addEventListener' : 'removeEventListener';
-		window[ handler ]( 'resize', this.throttledRefresh );
-		window[ handler ]( 'scroll', this.throttledRefresh, true );
-
-		/*
-		 * There are sometimes we need to reposition or resize the popover that are not
-		 * handled by the resize/scroll window events (i.e. CSS changes in the layout
-		 * that changes the position of the anchor).
-		 *
-		 * For these situations, we refresh the popover every 0.5s
-		 */
-		if ( isActive ) {
-			this.autoRefresh = setInterval( this.refreshOnAnchorMove, 500 );
-		} else {
-			clearInterval( this.autoRefresh );
+	};
+	useEffect( refreshAnchorRect, [ anchorRect, getAnchorRect ] );
+	useEffect( () => {
+		if ( ! anchorRect ) {
+			/*
+			* There are sometimes we need to reposition or resize the popover that are not
+			* handled by the resize/scroll window events (i.e. CSS changes in the layout
+			* that changes the position of the anchor).
+			*
+			* For these situations, we refresh the popover every 0.5s
+			*/
+			const intervalHandle = setInterval( refreshAnchorRect, 500 );
+
+			return () => clearInterval( intervalHandle );
 		}
-	}
+	}, [ anchorRect ] );
 
-	throttledRefresh( event ) {
-		window.cancelAnimationFrame( this.rafHandle );
-		if ( event && event.type === 'scroll' && this.contentNode.current.contains( event.target ) ) {
+	// Content size
+	const [ contentSize, setContentSize ] = useState( null );
+	useEffect( () => {
+		const contentRect = contentRef.current.getBoundingClientRect();
+		setContentSize( {
+			width: contentRect.width,
+			height: contentRect.height,
+		} );
+		setIsReadyToAnimate( true );
+	}, [] );
+
+	// Compute the position
+	const [ popoverPosition, setPopoverPosition ] = useState( {
+		popoverLeft: null,
+		popoverTop: null,
+		yAxis: 'top',
+		xAxis: 'center',
+		contentHeight: null,
+		contentWidth: null,
+		isMobile: false,
+	} );
+	const refreshPopoverPosition = () => {
+		if ( ! anchor || ! contentSize ) {
 			return;
 		}
-		this.rafHandle = window.requestAnimationFrame( this.refresh );
-	}
 
-	/**
-	 * Calling refreshOnAnchorMove
-	 * will only refresh the popover position if the anchor moves.
-	 */
-	refreshOnAnchorMove() {
-		const anchorRect = this.getAnchorRect( this.anchorNode.current );
-		const didAnchorRectChange = ! isShallowEqual( anchorRect, this.anchorRect );
-		if ( didAnchorRectChange ) {
-			this.anchorRect = anchorRect;
-			this.computePopoverPosition( this.state.popoverSize, anchorRect );
+		const newPopoverPosition = computePopoverPosition(
+			anchor,
+			contentSize,
+			position,
+			expandOnMobile
+		);
+
+		if (
+			popoverPosition.yAxis !== newPopoverPosition.yAxis ||
+				popoverPosition.xAxis !== newPopoverPosition.xAxis ||
```

Comment:
Shouldn't need to be indented more than the previous line.

Reply:
```suggestion
			popoverPosition.xAxis !== newPopoverPosition.xAxis ||
```